### ISSUE TITLE
moving past "Junior" + pizza toppings

### ISSUE THREAD

**Issue by johnakers** [r00k/AMA#43]
Hi! :smile: 

I have two questions.

1) Favorite pizza topping(s)? :pizza: 

2) I recently listened to [this](http://giantrobots.fm/63) episode of Giant Robots... and had a couple questions. (Also, thoroughly enjoy the podcast, particularly the functional programming stuff).

Currently, I'm a Junior Ruby / Rails developer for a super tiny startup in San Francisco (though not from SF). Its my first Software Engineering gig, ever. I've made a gem and done some personal projects / hackathons and tutor at a local bootcamp. The issue I have is... 

Where I work, I am really the only web dev person (we do mainly server side stuff / DevOps stuff).  I've learned a fair bit in that realm (Mongo / Docker / AWS / etc.) but to continue my Ruby and Rails learning, I don't really have anybody to do code review or really evaluate me in 'that' space.

I ask as I would like to be considered just a Ruby / Rails Developer and move past the Junior monicker, I'm not really sure as to the things I 'SHOULD' know, aside from what I think is cool and pursuing those.

I could go on... but I've already typed out a lot :confused: 

_TL:DR... How do I move past the "Junior" label / What skills are need to progress and just become a good Ruby/Rails developer?_

**Comment by queuebit** [comment]
I am pretty sure you/I are who http://upcase.com/ was created for.

**Comment by r00k** [comment]
> TL:DR... How do I move past the "Junior" label / What skills are need to progress and just become a good Ruby/Rails developer?

@queuebit is right. This is exactly why we made Upcase.

That said, here are a few core things:
- Practice TDD.
- Master a good editor (vim is my choice, emacs is also fine, the rest don't come close).
- Learn a good handful of patterns, code smells, and refactorings.
- Read Clean Code and Refactoring.

**Comment by johnakers** [comment]
Thanks for the response.

Started Upcase (but in the midst of working on weekends + personal project)... looking forward to jumping more into it.

And pizza toppings :question:

**Comment by r00k** [comment]
Hah!

Ideal: garlic, pepperoni, and spinach.

**Comment by r00k** [comment]
Glad to have you on Upcase, btw.

Let me know if you want to jump on a Skype call to discuss your learning.

**Comment by johnakers** [comment]
Definitely. Thanks :smile:
